test/export/test_export.py

```diff
@@ -1672,6 +1672,94 @@ def forward(self, x):
         res = ep.module()(ref_x)
         self.assertEqual(res, ref_out)
 
+    @testing.expectedFailureLegacyExportNonStrict  # Old export doesn't work with subclasses
+    @testing.expectedFailureLegacyExportStrict  # Old export doesn't work with subclasses
+    @testing.expectedFailureSerDerNonStrict  # builtins.getattr is not supported  T211130564
+    @testing.expectedFailureSerDer  # builtins.getattr is not supported  T211130564
+    def test_subclass_nested_attr_access(self):
+        from torch.testing._internal.two_tensor import TwoTensor
+
+        class Foo(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.p1 = torch.nn.Parameter(torch.ones(3, 4))
+                self.p2 = torch.nn.Parameter(
+                    TwoTensor(
+                        TwoTensor(torch.ones(3, 4), torch.ones(3, 4)),
+                        TwoTensor(torch.ones(3, 4), torch.ones(3, 4)),
```

Comment:
A few more testing ideas:

(1) a test where you have `TwoTensor(ConstantExtraMetadataTensor, ConstantExtraMetadataTensor)`, to test that we are properly proxying getattrs on subclass types that don't show up in the top-level arguments

(2) a test with a subclass that has more complicated constant metadata (e.g. a NamedTuple, to be a bit more representative of [Float8](https://github.com/pytorch/ao/blob/main/torchao/float8/float8_tensor.py#L46)). Mostly to make sure that something reasonable happens today, and also that we can augment later when we get constructor support (see comment below)

Reply:
Discussed offline. (2) won't actually be tested comprehensively given current state (need an operator or subclass constructor that consumes complicated metadata).

Reply:
Tbh it still seems worth testing to make sure something reasonable happens. e.g.:

```
named_tuple_val = subclass_tensor.metadata
# `named_tuple_val.foo` should get inlined as a constant
return subclass_tensor.elem * named_tuple_val.foo
```